googet.go

```diff
@@ -98,11 +107,18 @@ func unmarshalRepoFile(p string) ([]repoFile, error) {
 	// Both repoFile and []repoFile are valid for backwards compatibilty.
 	var rf repoFile
 	if err := yaml.Unmarshal(b, &rf); err == nil && rf.URL != "" {
+		rf.fileName = p
 		return []repoFile{rf}, nil
 	}
 
 	var rfs []repoFile
-	return rfs, yaml.Unmarshal(b, &rfs)
+	if err := yaml.Unmarshal(b, &rfs); err != nil {
+		return nil, err
+	}
+	for i := range rfs {
+		rfs[i].fileName = p
```

Comment:
Assigning fileName to the same thing for every element in rfs seems odd. Can you clarify why this is necessary?